utils/tracer_client.py

```diff
@@ -16,3 +18,176 @@ def send_rpc_and_wait_response(self, method_name, params, req_type=None, timeout
 
     def send_rpc(self, method, params, req_type=None):
         return self.tracer_api.send_rpc(method=method, params=params, req_type=req_type)
+
+    def debug_trace_call(self, tx_data: TxData):
+        params = [
+            {
+                "to": tx_data["to"],
+                "from": tx_data["from"],
+                "gas": hex(tx_data["gas"]),
+                "gasPrice": hex(tx_data["gasPrice"]),
+                "value": hex(tx_data["value"]),
+                "data": "0x" + tx_data["input"].hex(),
+            },
+            hex(tx_data["blockNumber"]),
+        ]
+        response = self.send_rpc_and_wait_response("debug_traceCall", params)
+
+        return response
+
+    def debug_trace_transaction(
+        self,
+        tx_hash: str,
+        tracer_type: str = "",
+        with_log=False,
+        only_top_call=False,
+        wait_response: int = 120,
+    ):
+        params = [tx_hash]
+        if tracer_type or with_log or only_top_call:
+            cfg = {"withLog": with_log, "OnlyTopCall": only_top_call}
+            params.append({"tracer": tracer_type, "tracerConfig": cfg})
+
+        return self.send_rpc_and_wait_response("debug_traceTransaction", params, timeout_sec=wait_response)
+
+    def debug_trace_block_by_number(
+        self,
+        block_number: str,
+        tracer_type: str = "",
+        with_log=False,
+        only_top_call=False,
+    ):
+
+        params = [block_number]
+        if tracer_type or with_log or only_top_call:
+            cfg = {"withLog": with_log, "OnlyTopCall": only_top_call}
+            params.append({"tracer": tracer_type, "tracerConfig": cfg})
+
+        response = self.tracer_api.send_rpc("debug_traceBlockByNumber", params)
+        return response
+
+    def debug_trace_block_by_hash(
+        self,
+        block_hash: str,
+        tracer_type: str = "",
+        with_log=False,
+        only_top_call=False,
+    ):
+        params = [block_hash]
+
+        if tracer_type or with_log or only_top_call:
+            cfg = {"withLog": with_log, "OnlyTopCall": only_top_call}
+            params.append({"tracer": tracer_type, "tracerConfig": cfg})
+
+        response = self.tracer_api.send_rpc("debug_traceBlockByHash", params)
+        return response
+
+    def debug_get_raw_header_by_block(self, block: str):
+        """block - block number or block hash in hex format"""
+        params = [block]
+        response = self.send_rpc_and_wait_response("debug_getRawHeader", params)
+        return response
+
+    def debug_get_raw_header_by_block_hash(self, block_hash: str):
```

Comment:
same story: block_number and block_hash have the same arg type and it can be one method